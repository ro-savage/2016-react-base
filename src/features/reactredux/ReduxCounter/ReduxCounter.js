import React from 'react'

import Button from '../../../components/Button/Button'

const counter = (props) => {
  return (
    <div>
      <span>Counter: {props.counter}</span>
      <br />
      <Button onClick={props.increase}>Up!</Button>
      <Button onClick={props.decrease}>Down!</Button>
      <Button onClick={props.increaseAsync}>Async!</Button>
    </div>
  )
}

counter.propTypes = {
  counter: React.PropTypes.number,
  increase: React.PropTypes.func,
  decrease: React.PropTypes.func,
  increaseAsync: React.PropTypes.func,
}

export default counter
