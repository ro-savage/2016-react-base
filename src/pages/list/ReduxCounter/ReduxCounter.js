import React from 'react'

const counter = (props) => {
  return (
    <div>
      <span>Counter: {props.counter}</span>
      <br />
      <button onClick={props.increase}>Up!</button>
      <button onClick={props.decrease}>Down!</button>
    </div>
  )
}

counter.propTypes = {
  counter: React.PropTypes.string,
  increase: React.PropTypes.func,
  decrease: React.PropTypes.func,
}

export default counter
