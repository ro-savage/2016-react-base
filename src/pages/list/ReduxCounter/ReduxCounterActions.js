export const increaseCounter = () => ({
  type: 'INCREMENT',
})

export const decreaseCounter = () => ({
  type: 'DECREMENT',
})

export const increaseCounterAsync = () => ({
  type: 'INCREMENT_ASYNC',
})
