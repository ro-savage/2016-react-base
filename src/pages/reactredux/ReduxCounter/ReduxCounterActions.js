export const INCREASE_COUNTER = '@@counter/INCREASE_COUNTER'
export const DECREMENT_COUNTER = '@@counter/DECREMENT_COUNTER'
export const INCREMENT_ASYNC = '@@counter/INCREMENT_ASYNC'

export const increaseCounter = () => ({
  type: INCREASE_COUNTER,
})

export const decreaseCounter = () => ({
  type: DECREMENT_COUNTER,
})

export const increaseCounterAsync = () => ({
  type: INCREMENT_ASYNC,
})
