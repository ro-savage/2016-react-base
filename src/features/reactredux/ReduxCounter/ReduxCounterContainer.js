import { connect } from 'react-redux'

import ReduxCounter from './ReduxCounter'
import { increaseCounter, decreaseCounter, increaseCounterAsync } from './ReduxCounterActions'

const mapStateToProps = ({ reduxCounter }) => ({
  counter: reduxCounter.counter,
})

const mapDispatchToProps = dispatch => ({
  increase: () => dispatch(increaseCounter()),
  decrease: () => dispatch(decreaseCounter()),
  increaseAsync: () => dispatch(increaseCounterAsync()),
})

const ReduxCounterContainer = connect(mapStateToProps, mapDispatchToProps)(ReduxCounter)

export default ReduxCounterContainer
