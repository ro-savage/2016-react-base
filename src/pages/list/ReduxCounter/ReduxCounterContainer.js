import { connect } from 'react-redux'

import ReduxCounter from './ReduxCounter'
import { increaseCounter, decreaseCounter } from './ReduxCounterActions'

const mapStateToProps = ({ reduxCounter }) => ({
  counter: reduxCounter.counter,
})

const mapDispatchToProps = dispatch => ({
  increase: () => dispatch(increaseCounter()),
  decrease: () => dispatch(decreaseCounter()),
})

const ReduxCounterContainer = connect(mapStateToProps, mapDispatchToProps)(ReduxCounter)

export default ReduxCounterContainer
