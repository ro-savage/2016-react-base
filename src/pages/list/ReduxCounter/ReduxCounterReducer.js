import * as actions from './ReduxCounterActions'

const reducerState = {
  counter: 0,
}

const ReduxCounterReducer = (state = reducerState, action) => {
  switch (action.type) {
    case actions.INCREASE_COUNTER:
      return {
        ...state,
        counter: state.counter + 1,
      }
    case actions.DECREMENT_COUNTER:
      return {
        ...state,
        counter: state.counter - 1,
      }
    default:
      return state
  }
}

export default ReduxCounterReducer
