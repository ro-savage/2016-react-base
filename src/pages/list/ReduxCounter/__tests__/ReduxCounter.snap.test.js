import React from 'react'
import renderer from 'react-test-renderer'
import ReduxCounter from '../ReduxCounter'


test('ReactList changes when new item is added', () => {
  const component = renderer.create(
    <ReduxCounter counter={1} increase={() => {}} decrease={() => {}} increaseAsync={() => {}} />,
  )

  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
