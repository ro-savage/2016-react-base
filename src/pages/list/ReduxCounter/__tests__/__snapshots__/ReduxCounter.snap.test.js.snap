exports[`test ReactList changes when new item is added 1`] = `
<div>
  <span>
    Counter: 
    1
  </span>
  <br />
  <button
    className={undefined}
    onClick={[Function]}>
    Up!
  </button>
  <button
    className={undefined}
    onClick={[Function]}>
    Down!
  </button>
  <button
    className={undefined}
    onClick={[Function]}>
    Async!
  </button>
</div>
`;
